### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Groupoid.Subgroupoid

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Groupoid/Subgroupoid.lean [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+# Subgroupoid
+
+This file defines subgroupoids as `structure`s containing the subsets of arrows and their
+stability under composition and inversion.
+Also defined are:
+
+* containment of subgroupoids is a complete lattice;
+* images and preimages of subgroupoids under a functor;
+* the notion of normality of subgroupoids and its stability under intersection and preimage;
+* compatibility of the above with `groupoid.vertex_group`.
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Groupoid/Subgroupoid.lean:29]:
```suggestion
* compatibility of the above with `Groupoid.vertexGroup`.
```

File: Mathlib/CategoryTheory/Groupoid/Subgroupoid.lean [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+      suffices Map.Arrows φ hφ S (φ.obj c') (φ.obj c') (φ.map <| Groupoid.inv f ≫ γ ≫ f) by
+        simp only [inv_eq_inv, Functor.map_comp, Functor.map_inv] at this ; exact this
+      · constructor; apply Sn.conj f γS }
+#align category_theory.subgroupoid.is_normal_map CategoryTheory.Subgroupoid.isNormal_map
+
+end Hom
+
+section Thin
+
+/-- A subgroupoid `is_thin` if it has at most one arrow between any two vertices. -/
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Groupoid/Subgroupoid.lean:603]:
```suggestion
/-- A subgroupoid `IsThin` if it has at most one arrow between any two vertices. -/
```

**Comment by semorrison** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**